src/scanner/present.rs

```diff
@@ -392,17 +446,16 @@ async fn is_active_oracle_expired(
     let expiration = match active_oracle.expiration {
         Some(expiration) => expiration,
         None => {
-            let mut db_connection = db_connection_pool.get().context(format!(
-                "could not get database connection while trying to update oracle's expiration"
-            ))?;
-
             let expiration =
                 fetch_active_oracle_expiration(signer.clone(), active_oracle.address.0)
                     .await
                     .context(format!(
                         "could not fetch active oracle 0x{:x} expiration",
                         active_oracle.address.0
                     ))?;
+            let mut db_connection = db_connection_pool.get().context(format!(
```

Comment:
Should we drop the connection after the expiration update here?

Reply:
It shouldn't really be necessary here as the current scope expires right after the update, dropping the connection naturally